src/measure_theory/vector_measure.lean

```diff
@@ -815,4 +815,66 @@ end
 
 end vector_measure
 
+namespace signed_measure
+
+open vector_measure
+
+open_locale measure_theory
+
+/-- The underlying function for `signed_measure.to_measure`. -/
+def to_measure' (s : signed_measure α)
+  (i : set α) (hi₁ : measurable_set i) (hi₂ : 0 ≤[i] s)
+  (j : set α) (hj₁ : measurable_set j) : ℝ≥0∞ :=
+some ⟨s.restrict i j,
+begin
+  rw [s.restrict_apply hi₁ hj₁, ← s.restrict_apply hi₁ hj₁],
+  exact le_trans (by simp) (hi₂ j hj₁),
+end⟩
+
+/-- Given a signed measure `s` and a positive measurable set `i`, `to_measure`
+provides the measure, mapping measurable sets `j` to `s (i ∩ j)`. -/
+def to_measure (s : signed_measure α) (i : set α) (hi₁ : measurable_set i) (hi₂ : 0 ≤[i] s) :
+  measure α :=
+measure.of_measurable (s.to_measure' i hi₁ hi₂)
+  (by { simp_rw [to_measure', s.restrict_apply hi₁ measurable_set.empty,
+                 set.empty_inter i, s.empty], refl })
+  begin
+    intros f hf₁ hf₂,
+    have h₁ : ∀ n, measurable_set (i ∩ f n) := λ n, hi₁.inter (hf₁ n),
+    have h₂ : pairwise (disjoint on λ (n : ℕ), i ∩ f n),
+    { rintro n m hnm x ⟨⟨_, hx₁⟩, _, hx₂⟩,
+      exact hf₂ n m hnm ⟨hx₁, hx₂⟩ },
+    simp_rw [to_measure', s.restrict_apply hi₁ (measurable_set.Union hf₁),
+             set.inter_comm, set.inter_Union, s.of_disjoint_Union_nat h₁ h₂,
+             ennreal.some_eq_coe],
+    rw [nnreal.coe_tsum_of_nonneg, ennreal.coe_tsum],
```

Comment:
after this rewrite, you have three goals, so this should show up in your braces structure. Since there is something strange going on (with the same goal `∀ (n : ℕ), 0 ≤ ⇑s (i ∩ f n)` that you have to prove twice), you could have it as a 
` have H : ∀ (n : ℕ), 0 ≤ ⇑s (i ∩ f n)` before the rewrite, and then use it directly in the terms where you need it.